src/scipp/coords/rule.py

```diff
@@ -13,43 +13,34 @@
 from collections.abc import Callable, Iterable, Mapping
 from copy import copy
 from functools import partial
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Any, TypeVar
 
 from ..core import Variable
 from .coord import Coord
 
 if TYPE_CHECKING:
-    from typing import Protocol as _Protocol
-
-    # Importing CoordTable from coord_table.py would result in an import
-    # cycle because that module imports rule.py
-    # CoordTable is only needed for type annotations here,
-    # so a protocol is enough.
-    class _CoordProvider(_Protocol):
-        def consume(self, name: str) -> Coord:
-            pass
-
```

Comment:
Was the comment wrong, or what has changed?

Reply:
It wasn't wrong. But the new solution matches what we do in other places (https://github.com/scipp/scipp/blob/606929439a331ac81edc16e41bdcea44d85839b3/src/scipp/coords/rule.py#L21-L24) Import cycles are fine when type checking. And during runtime, we don't need to provide a precise type hint because we don't use any runtime type checker.